plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueStatConverter.java

```diff
@@ -0,0 +1,291 @@
+    {
+        ColumnStatisticsType type = ColumnStatisticsType.fromValue(catalogColumnStatisticsData.getType());
+        switch (type) {
+            case BINARY:
+                BinaryColumnStatisticsData catalogBinaryData = catalogColumnStatisticsData.getBinaryColumnStatisticsData();
+                OptionalLong maxColumnLengthOfBinary = OptionalLong.of(catalogBinaryData.getMaximumLength());
+                OptionalDouble averageColumnLengthOfBinary = OptionalDouble.of(catalogBinaryData.getAverageLength());
+                OptionalLong nullsCountOfBinary = fromMetastoreNullsCount(catalogBinaryData.getNumberOfNulls());
+                return createBinaryColumnStatistics(
+                        maxColumnLengthOfBinary,
+                        getTotalSizeInBytes(averageColumnLengthOfBinary, rowCount, nullsCountOfBinary),
+                        nullsCountOfBinary);
+
+            case BOOLEAN:
+                BooleanColumnStatisticsData catalogBooleanData = catalogColumnStatisticsData.getBooleanColumnStatisticsData();
+                return createBooleanColumnStatistics(
+                        OptionalLong.of(catalogBooleanData.getNumberOfTrues()),
+                        OptionalLong.of(catalogBooleanData.getNumberOfFalses()),
+                        fromMetastoreNullsCount(catalogBooleanData.getNumberOfNulls()));
+
+            case DATE:
+                DateColumnStatisticsData catalogDateData = catalogColumnStatisticsData.getDateColumnStatisticsData();
+                Optional<LocalDate> minOfDate = dateToLocalDate(catalogDateData.getMinimumValue());
+                Optional<LocalDate> maxOfDate = dateToLocalDate(catalogDateData.getMaximumValue());
+                OptionalLong nullsCountOfDate = fromMetastoreNullsCount(catalogDateData.getNumberOfNulls());
+                OptionalLong distinctValuesCountOfDate = OptionalLong.of(catalogDateData.getNumberOfDistinctValues());
+                return createDateColumnStatistics(minOfDate, maxOfDate, nullsCountOfDate, fromMetastoreDistinctValuesCount(distinctValuesCountOfDate, nullsCountOfDate, rowCount));
+
+            case DECIMAL:
+                DecimalColumnStatisticsData catalogDecimalData = catalogColumnStatisticsData.getDecimalColumnStatisticsData();
+                Optional<BigDecimal> minOfDecimal = glueDecimalToBigDecimal(catalogDecimalData.getMinimumValue());
+                Optional<BigDecimal> maxOfDecimal = glueDecimalToBigDecimal(catalogDecimalData.getMaximumValue());
+                OptionalLong distinctValuesCountOfDecimal = OptionalLong.of(catalogDecimalData.getNumberOfDistinctValues());
+                OptionalLong nullsCountOfDecimal = fromMetastoreNullsCount(catalogDecimalData.getNumberOfNulls());
+                return createDecimalColumnStatistics(minOfDecimal, maxOfDecimal, nullsCountOfDecimal, fromMetastoreDistinctValuesCount(distinctValuesCountOfDecimal, nullsCountOfDecimal, rowCount));
```

Comment:
Use braces. This will allow you to use shorter variable names. Like this:
```suggestion
            case DATE: {
                DateColumnStatisticsData data = catalogColumnStatisticsData.getDateColumnStatisticsData();
                Optional<LocalDate> min = dateToLocalDate(data.getMinimumValue());
                Optional<LocalDate> max = dateToLocalDate(data.getMaximumValue());
                OptionalLong nullsCount = fromMetastoreNullsCount(data.getNumberOfNulls());
                OptionalLong numberOfDistinctValues = OptionalLong.of(data.getNumberOfDistinctValues());
                return createDateColumnStatistics(min, max, nullsCount, fromMetastoreDistinctValuesCount(numberOfDistinctValues, nullsCount, rowCount));
            }
            case DECIMAL: {
                DecimalColumnStatisticsData data = catalogColumnStatisticsData.getDecimalColumnStatisticsData();
                Optional<BigDecimal> min = glueDecimalToBigDecimal(data.getMinimumValue());
                Optional<BigDecimal> max = glueDecimalToBigDecimal(data.getMaximumValue());
                OptionalLong numberOfDistinctValues = OptionalLong.of(data.getNumberOfDistinctValues());
                OptionalLong nullsCount = fromMetastoreNullsCount(data.getNumberOfNulls());
                return createDecimalColumnStatistics(min, max, nullsCount, fromMetastoreDistinctValuesCount(numberOfDistinctValues, nullsCount, rowCount));
            }
```